alibi_detect/ad/model_distillation.py

```diff
@@ -0,0 +1,220 @@
+import logging
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.losses import kld, categorical_crossentropy
+from typing import Callable, Dict, Tuple, Union
+from alibi_detect.models.trainer import trainer
+from alibi_detect.models.losses import loss_distillation
+from alibi_detect.utils.prediction import predict_batch
+from alibi_detect.base import (BaseDetector, FitMixin, ThresholdMixin,
+                               adversarial_prediction_dict)
+
+logger = logging.getLogger(__name__)
+
+
+class ModelDistillation(BaseDetector, FitMixin, ThresholdMixin):
+
+    def __init__(self,
+                 threshold: float = None,
+                 distilled_model: tf.keras.Model = None,
+                 model: tf.keras.Model = None,
+                 loss_type: str = 'kld',
+                 temperature: float = 1.,
+                 data_type: str = None
+                 ) -> None:
+        """
+        Model distillation concept drift and adversarial detector.
+
+        Parameters
+        ----------
+        threshold
+            Threshold used for score to determine adversarial instances.
+        distilled_model
+            A tf.keras model to distill.
+        model
+            A trained tf.keras classification model.
+        loss_type
+            Loss for distillation. Supported: 'kld', 'xent'
+        temperature
+            Temperature used for model prediction scaling.
+            Temperature <1 sharpens the prediction probability distribution.
+        data_type
+            Optionally specifiy the data type (tabular, image or time-series). Added to metadata.
+        """
+        super().__init__()
+
+        if threshold is None:
+            logger.warning('No threshold level set. Need to infer threshold using `infer_threshold`.')
+
+        self.threshold = threshold
+        self.model = model
+        for layer in self.model.layers:  # freeze model layers
+            layer.trainable = False
+
+        # check if model can be loaded, otherwise initialize AE model
```

Comment:
Comment not really relevant b/c model not loaded. Also is not an AE. Would just delete comment.